src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -949,6 +941,79 @@ public void testConsecutiveQueries() throws SQLException {
+                int newValue = 2012;
+                stmt.setMaxFieldSize(newValue);
+                actual = stmt.getMaxFieldSize();
+                assertEquals(actual, newValue, "MaxFieldSize() : set/get problem");
+
+                // execute a statement with this new max field size
+                stmt.execute("SELECT @@VERSION AS 'SQL Server Version'");
+
+                /*
+                 * Set a new value greater than MAX_VALUE, and then get the modified value SQL Server only supports
+                 * integer limits for setting max rows If the value MAX_VALUE + 1 is accepted, throw exception
+                 */
+                try {
+                    newValue = Integer.MAX_VALUE + 1;
+                    stmt.setMaxFieldSize(newValue);
+                    fail(TestResource.getResource("R_expectedFailPassed"));
+                } catch (Exception e) {
+                    assertTrue(e.getMessage().matches(TestUtils.formatErrorMsg("R_invalidLength")));
+                }
+            }
+        }
+
+        @Test
+        public void testMaxRows() throws Exception {
+            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+
+                // Default value should return zero
+                long actual = stmt.getMaxRows();
+                assertEquals(actual, (long) 0, "getMaxRows():" + TestResource.getResource("R_incorrectDefault"));
+
+                // Set a new value less than MAX_VALUE, and then get the modified value
+                int newValue = 2012;
+                stmt.setMaxRows(newValue);
+                actual = stmt.getMaxRows();
+                assertEquals(actual, newValue, "MaxRows() : set/get problem");
+
+                // execute a statement with this new max field size
+                stmt.execute("SELECT @@VERSION AS 'SQL Server Version'");
+
+                /*
+                 * Set a new value greater than MAX_VALUE, and then get the modified value SQL Server only supports
+                 * integer limits for setting max rows If the value MAX_VALUE + 1 is accepted, throw exception
+                 */
+                try {
+                    newValue = Integer.MAX_VALUE + 1;
+                    stmt.setMaxRows(newValue);
+                    fail(TestResource.getResource("R_expectedFailPassed"));
+                } catch (Exception e) {
+                    assertTrue(e.getMessage().matches(TestUtils.formatErrorMsg("R_invalidRowcount")));
+                }
+
+                // Set a negative value. If negative is accepted, throw exception
+                try {
+                    stmt.setMaxRows(-2012);
+                    fail(TestResource.getResource("R_expectedFailPassed"));
+                } catch (Exception e) {
+                    assertTrue(
+                            e.getMessage()
+                                    .contains("The maximum row count -2,012 for a result set must be non-negative."),
```

Comment:
Why are we hardcoding error message here and not using `R_invalidRowcount` from SQLServerResource?